nixos/modules/security/sudo.nix

```diff
@@ -223,7 +223,7 @@ in
 
     environment.systemPackages = [ sudo ];
 
-    security.pam.services.sudo = { sshAgentAuth = true; };
+    security.pam.services.sudo = lib.optionalAttrs config.security.pam.enableSSHAgentAuth { sshAgentAuth = true; };
```

Comment:
I think it probably makes more sense to move this definition to the `pam.nix` module instead, then option definition and use is in the same file.